pep-0484.txt

```diff
@@ -276,6 +276,33 @@ implemented by deferring to ``isinstance(x, collections.abc.Callable)``.
 However, ``isinstance(x, typing.Callable[...])`` is not supported.
 
 
+Decorators
+----------
+
+Decorators can modify the types of the functions or classes they
+decorate. To annotate a decorated declaration with the eventual
+decorated type, put a type comment on the line with the decorator::
+
+  from typing import ContextManager, Generator
+  from contextlib import contextmanager
+
+  class DatabaseSession: ...
+
+  @contextmanager  # type: Callable[[str], ContextManager[DatabaseSession]]
+  def session(url: str) -> Generator[DatabaseSession, None, None]:
+      s = DatabaseSession(url)
+      try:
+          yield s
+      finally:
+          s.close()
+
+If both the decorator and the function being decorated have type
```

Comment:
@sixolet I would be happy with just extending "decorator" to "the definition of the decorator" or "the decorator function".